### PR TITLE
Add google.auth.transport and google.auth.exceptions

### PR DESCRIPTION
**Pull request by jonparrott**
@dhermes you've already mostly review this.

### CODE REVIEW

File: google/auth/exceptions.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Exceptions used in the google.auth package."""
+
+
+class GoogleAuthError(Exception):
+    """Base class for all google.auth errors."""
+    pass
```
Comment (dhermes) [inline google/auth/exceptions.py]: No pass needed.

File: google/auth/exceptions.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+
+class TransportError(Exception):
+    """Used to indicate an error occurred during an HTTP request."""
+    pass
+
+
+class RefreshError(GoogleAuthError):
+    """Used to indicate that an error occurred while refreshing the
+    credentials' access token."""
```
Comment (dhermes) [inline google/auth/exceptions.py]: You already know I'm :-1: on a single-line description taking up multiple lines.

File: google/auth/transport/__init__.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+    @abc.abstractproperty
+    def headers(self):
+        """Mapping: The HTTP response headers."""
+        raise NotImplementedError('headers must be implemented.')
+
+    @abc.abstractproperty
+    def data(self):
+        """bytes: The response body."""
+        raise NotImplementedError('headers must be implemented.')
```
Comment (dhermes) [inline google/auth/transport/__init__.py]: s/headers/data

File: google/auth/exceptions.py (hunk 1 above)
Comment (jonparrott) [inline google/auth/exceptions.py]: Done

File: google/auth/exceptions.py (hunk 2 above)
Comment (jonparrott) [inline google/auth/exceptions.py]: I'm so on the fence about this one. So many projects break it and I think clarity is more important than dogma.

File: google/auth/transport/__init__.py (hunk 3 above)
Comment (jonparrott) [inline google/auth/transport/__init__.py]: Done, good catch.

**Comment by jonparrott** [comment]
> No tests?

Everything here is abstract.

**Pull request merged**